### ISSUE TITLE
Wrong log when using `autocreate`

### ISSUE THREAD

**Issue by pchico83** [okteto/okteto#2719]
**Describe the bug**
We are logging this warning when using a dev container with `autocreate`:
```
 !  could not add dev annotations due to: the application 'xxx' referred by your okteto manifest doesn't exist
 ```
 and this is fine, we shouldn't warn the user

**Comment by pchico83** [comment]
This is throwing an error in 2.3.0, but not in 2.0.2

**Comment by jmacelroy** [comment]
I am unable to reproduce this and will probably need more information in able to do so. I created a v2 manifest and compose for go-getting-started at https://github.com/jmacelroy/go-getting-started/blob/autocreate-with-2.0-manifest/okteto.yml

I am always successfully deploying, using `okteto up`, with no deployment existing and `autocreate: true` set.

I tried with the manifest in the branch, the manifest in the branch with `build` and `deploy` sections commented out, and with different names in compose file `services` section and the okteto manifest `dev` target. 

In all of those cases 2.3.0 worked with no errors shown. Is it possible there's an interaction issue with certain okteto server versions? I was using cloud.okteto.com
